restli-client/src/main/java/com/linkedin/restli/client/RestLiResponseException.java

```diff
@@ -64,74 +66,84 @@ public RestLiResponseException(RestResponse rawResponse,
                                  Throwable cause)
   {
     super(rawResponse, cause);
-    _status = rawResponse.getStatus();
-    _errorResponse = errorResponse;
-    _decodedResponse = decodedResponse;
+    _status = Optional.of(rawResponse.getStatus());
+    _errorResponse = Optional.ofNullable(errorResponse);
```

Comment:
From pegasus codebase, we have guaranteed that ErrorResponse passed in cannot be null, which can be seen from ExceptionUtil.getErrorResponse() method. I am curious in what case you encountered a NULL passed in for _errorResponse?